core/common/src/main/java/alluxio/conf/AlluxioProperties.java

```diff
@@ -58,6 +59,11 @@
   /** Map of property sources. */
   private final ConcurrentHashMap<PropertyKey, Source> mSources = new ConcurrentHashMap<>();
 
+  private final Version mVersion = new Version(() -> keySet().stream()
```

Comment:
What's the concurrency guarantee around this? Do all users of `AlluxioProperties` ensure thread safety?

Reply:
made `Version` thread safe.